### PR TITLE
Fix table editor unable to update PK after renaming table

### LINKED ISSUE #19341: Unable to update table PK via dashboard after renaming table
# Bug report

## Describe the bug

Opening this issue because I ran into it again today #5549

After renaming a table, the _pkey constraint uses the old table name. This causes a problem if you try to change the primary key field.

For example, a table originally named "reporttype_report" was renamed to "report_reporttype".

When trying to change the primary key, It gives an error "constraint "report_reporttype_pkey" of relation "report_reporttype" does not exist" because the constraint is still named "reporttype_report_pkey"

## To Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:

1. create a table
2. rename it in the admin console
3. add a new column
4. try to add the new column to the primary key

## Expected behavior
Should succeed. Instead you have to go find the constraint name and manually ALTER the table.

## System information
OS: macOS
Browser: Chrome

### PR DESCRIPTION
**Pull request by joshenlim**
Resolves https://github.com/supabase/supabase/issues/19341

### CODE REVIEW

File: apps/studio/components/interfaces/TableGridEditor/SidePanelEditor/SidePanelEditor.utils.tsx [hunk 1]
```diff
@@ -266,11 +268,13 @@ export const createColumn = async ({
   connectionString,
   payload,
   selectedTable,
+  primaryKey,
 }: {
   projectRef: string
   connectionString: string | undefined
   payload: CreateColumnPayload
   selectedTable: PostgresTable
+  primaryKey?: Constraint
```
Comment (ivasilov) [inline apps/studio/components/interfaces/TableGridEditor/SidePanelEditor/SidePanelEditor.utils.tsx:277]: Why pass in the primary key when it's available on `selectedTable.primary_keys`?

File: apps/studio/data/database/constraints-query.ts [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    {
+      projectRef,
+      connectionString,
+      sql: getTableConstraints({ schema, table }),
+      queryKey: ['table-constraints'],
+    },
+    {
+      enabled: typeof schema !== 'undefined' && typeof table !== 'undefined',
+      select(data) {
+        return ((data as any)?.result ?? []).map((constraint: Constraint) => constraint)
```
Comment (alaister) [inline apps/studio/data/database/constraints-query.ts:64]: Does this do the same thing? The map isn't doing anything apart from adding types
```suggestion
        return ((data as any)?.result ?? []) as Constraint[]
```

**Review by alaister — APPROVED** [review]
LGTM!

File: apps/studio/data/database/constraints-query.ts (hunk 2 above)
Comment (joshenlim) [reply apps/studio/data/database/constraints-query.ts:64]: not sure why but TS is complaining 😕 

image

Comment (alaister) [reply apps/studio/data/database/constraints-query.ts:64]: hmm maybe the whole map can just be removed entirely then?

Comment (joshenlim) [reply apps/studio/data/database/constraints-query.ts:64]: unless we rewrite it like this?

```export const useTableConstraintsQuery = (
  { projectRef, connectionString, schema, table }: TableConstraintsVariables,
  options: UseQueryOptions<ExecuteSqlData, TableConstraintsError, Constraint[]> = {}
) => {
  return useExecuteSqlQuery(
    {
      projectRef,
      connectionString,
      sql: getTableConstraints({ schema, table }),
      queryKey: ['table-constraints'],
    },
    {
      enabled: typeof schema !== 'undefined' && typeof table !== 'undefined',
      select(data) {
        return ((data as any)?.result ?? []) as Constraint[]
      },
      ...options,
    }
  )
}```

Comment (joshenlim) [reply apps/studio/data/database/constraints-query.ts:64]: > hmm maybe the whole map can just be removed entirely then?

oh 🤦 that worked hahaha

**Pull request merged**